pkg/apis/core/v1/defaults.go

```diff
@@ -157,6 +159,27 @@ func SetDefaults_Pod(obj *v1.Pod) {
 				}
 			}
 		}
+		if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+			// For normal containers, set resize restart policy to default value (RestartNotRequired), if not specified.
+			resizePolicySpecified := make(map[v1.ResourceName]bool)
+			for _, p := range obj.Spec.Containers[i].ResizePolicy {
+				resizePolicySpecified[p.ResourceName] = true
+			}
+			if _, found := resizePolicySpecified[v1.ResourceCPU]; !found {
```

Comment:
Should this have policies for resources my pod did not specify?  I'm asking not telling.  It's not clear to me.

I was thinking something like:

```
for each resource in requests {
    ensure a policy exists
}
```

We then need to ask - should it be OK to specify policies for resources that don't exist?  Probably - we can't really tell if the user put it there or we did, so best to leave them.

Reply:
I know we set default values for fields, but those fields are all scalar in nature, whereas resources are associative (and extensible).

What I don't like about adding them here is that we are hard-coding, DEEEEEP inside the stack, the set of resources we support.

WRT patching to add resources, that should be OK - this logic runs on create and update.  If you start with no resources defined and then patch some in, the simple loop I sketched above still kicks in.  It doesn't handle the REMOVAL of a resource - is that important?

Reply:
My hunch is neither add nor remove would be really important. If users need a certain amount of cpu and/or memory, they will specify it, and with in-place resize, they will expect that the spec gets updated by VPA with usage.

Defaulting resize policy only for specified resources seems reasonable given the associativity point, I'll make the change shortly.